File: tests/clCreateCommandQueue.py
#!/usr/bin/env python
import sys, re

sys.path.append('../python')
from prof_parser import prof_parse
from prof_parser import ptr_regex

# Test info.
call = 'clCreateCommandQueue'
_id  = ''
print '%s%s' % (call, _id)

# Parse test source.
source = {}
with open(call + _id + '.cpp', 'r') as f:
    source['text'] = f.read()
    source['context'] = re.search('\(cl_context\) (?P<context>%s)' % ptr_regex, source['text']).group('context')
    source['device'] = re.search('\(cl_device_id\) (?P<device>%s)' % ptr_regex, source['text']).group('device')
    source['properties'] = re.search('\(cl_command_queue_properties\) (?P<props>\d*)', source['text']).group('props')
    source['errcode_ret'] = re.search('\(cl_int \*\) (?P<errcode_ret>%s)' % ptr_regex, source['text']).group('errcode_ret')
    # The following should match the assert statement.
    source['queue'] = re.search('\(cl_command_queue\) (?P<queue>%s)' % ptr_regex, source['text']).group('queue')

# Read from stdin (via pipe).
output = sys.stdin.read()
print 'OUTPUT'
print output

result = prof_parse(output)[0]
print 'RESULT'
print result
print

status = True
status &= (source['context'] == result['context'])
status &= (source['device'] == result['device'])
status &= (source['properties'] == result['properties'])
status &= (source['errcode_ret']  == result['errcode_ret'])

print '%s%s: %s' % (call, _id, 'PASSED' if status else 'FAILED')
print

File: tests/clCreateCommandQueue.cpp
#include <CL/opencl.h>
#include <cassert>

int main()
{
    cl_context context = (cl_context) 0x01234567;
    cl_device_id device = (cl_device_id) 0x12345678;
    cl_command_queue_properties properties = (cl_command_queue_properties) 0;
    cl_int * errcode = (cl_int *) 0x23456789;
    
    cl_command_queue queue = clCreateCommandQueue(context, device, properties, errcode);
    assert(((cl_command_queue) 0x00000000 == queue));

    return 0;
}
